syntax = "proto2";
import "goby/middleware/protobuf/transporter_config.proto";
import "google/protobuf/descriptor.proto";

package goby.middleware.protobuf;

message SerializerTransporterKey
{
    required int32 marshalling_scheme = 1;
    required string type = 2;
    required string group = 3;
    optional TransporterConfig cfg = 4;
}


message SerializerTransporterMessage
{
    required SerializerTransporterKey key = 1;
    optional bytes data = 4;
}

// message DCCLSubscription
// {
//     required int32 dccl_id = 1;
//     required int32 group = 2;
//     // defines the DCCL message for the Portal (edge)
//     required string protobuf_name = 3;
//     repeated google.protobuf.FileDescriptorProto file_descriptor = 4;

//     optional TransporterConfig cfg = 5;
// }

message DCCLPacket
{
    required int32 dccl_id = 1;
    required bytes data = 2;
}

message DCCLForwardedData
{
    repeated DCCLPacket frame = 1;
}
